cekit-helpers/cekit-cache-server/src/main/java/org/kie/cekit/cacher/utils/CacherUtils.java

```diff
@@ -70,6 +74,49 @@ public void cleanTmpFiles() {
         }
     }
 
+    /**
+     * Clean RHPAM and RHDM nightly artifacts, keep only latest 3 nightly builds
+     * run once a day.
+     */
+    @Scheduled(every = "24h", delay = 12, delayUnit = TimeUnit.HOURS)
+    public void cleanOldProductNightlyArtifacts() throws IOException, InterruptedException {
+        log.fine("Trying to identify the latest nightly build based on filesystem artifacts to delete, this could take a while...");
+        try {
+            log.fine("Walking through data dir searching for old nightly product builds...");
+
+            Map<File, LocalDate> nightlyBuildArtifacts = new HashMap<>();
+            List<LocalDate> foundDates = new ArrayList<>();
+            Files.walk(Paths.get(cacherProperties.getCacherArtifactsDir()))
+                    .map(Path::toFile)
+                    .filter(file -> cacherProperties.buildDatePattern.matcher(file.getName()).find())
+                    .filter(file -> file.getName().endsWith(".zip"))
+                    .forEach(file -> {
+                        Matcher test = cacherProperties.buildDatePattern.matcher(file.getName());
+                        if (test.find()) {
+                            LocalDate d = LocalDate.parse(test.group(0), cacherProperties.formatter);
+                            // collect all files that matches the build date pattern
+                            nightlyBuildArtifacts.put(file, d);
+                            // collect all build dates to get the newst one to be the base date.
+                            if (!foundDates.contains(d)) {
+                                foundDates.add(d);
+                            }
+                        }
+                    });
+
+            LocalDate mostRecentNightlyBuild = Collections.max(foundDates);
+            log.fine("Latest nightly build date is -> " + mostRecentNightlyBuild);
+
+            for (Map.Entry<File, LocalDate> entry : nightlyBuildArtifacts.entrySet()) {
+                if (entry.getValue().plusDays(3).isBefore(mostRecentNightlyBuild)) {
+                    log.fine("File [" + entry.getKey() + " is 3 days older than the latest build date [" + mostRecentNightlyBuild + "]. Deleting...");
+                    Files.delete(entry.getKey().toPath());
```

Comment:
In comment for this method is "keep only latest 3 nightly builds". This may not be true as we are checking only latest build and there may not be nightly build for 3 days.
I think it should be OK to keep latest nightly and 3 days old builds - so only update of comment would be ok :)

Reply:
updated.